### PR TITLE
fix: support jinja templates

### PR DESCRIPTION
**Pull request by bkyryliuk**
### SUMMARY

1. Adds jinja support to the alert queries
2. Removes comment stripping for better transparency for the users & easier audit

Side change:
* extra test to cover multi line queries & comments, is not related to the fix

### TEST PLAN
* unit tests
* local and staging testing

### CODE REVIEW

**Review by villebro — COMMENTED** [review]
Can you add some more context to the description, it appears this is a combined fix (the multiline/comment problem) + feat (enable templating). Also, while I assume inline comments won't be a problem in this case as these queries aren't wrapped in a subquery, I would be interested in understanding the need for removing comment stripping.

File: superset/tasks/alerts/observer.py [hunk 1]
```diff
@@ -42,9 +42,9 @@ def observe(alert_id: int, session: Session) -> Optional[str]:
 
     value = None
 
-    parsed_query = ParsedQuery(sql_observer.sql)
-    sql = parsed_query.stripped()
-    df = sql_observer.database.get_df(sql)
+    tp = jinja_context.get_template_processor(database=sql_observer.database)
+    rendered_sql = tp.process_template(sql_observer.sql)
+    df = sql_observer.database.get_df(rendered_sql)
```
[the remark this answers is not in the document]
Comment (bkyryliuk) [reply superset/tasks/alerts/observer.py:47]: main reason is that we internally are logging the queries & doing their introspection. Users may be surprised to see modified queries.

File: tests/alerts_tests.py [hunk 2]
```diff
@@ -154,6 +154,27 @@ def test_alert_observer(setup_database):
...
+        -- comment
+        SELECT
+            1 -- comment
+        FROM test_table
+            WHERE first = 1
+        """,
+    )
+    observe(alert8.id, dbsession)
+    assert alert8.sql_observer[0].observations[-1].value == 1.0
+    assert alert8.sql_observer[0].observations[-1].error_msg is None
```
[the remark this answers is not in the document]
Comment (bkyryliuk) [reply tests/alerts_tests.py:170]: it was failing prior to other fixes however was working prior to this PR. I just wanted to make sure that this test case exists and we will be able to catch regressions

**Review by bkyryliuk — COMMENTED** [review]
> Can you add some more context to the description, it appears this is a combined fix (the multiline/comment problem) + feat (enable templating). Also, while I assume inline comments won't be a problem in this case as these queries aren't wrapped in a subquery, I would be interested in understanding the need for removing comment stripping.

this only fixes jinja, will add a better description
as for comment stripping - it is our preference to keep queries the way they are for simple audit and user transparency

**Review by villebro — APPROVED** [review]
LGTM

**Pull request merged**